aiverify-foundation/moonshot-ui
# Feature/add recipes to cookbook

## Description

Code changes enables 

- selecting recipes on recipes list screen and then adding them to an existing or new cookbook.
- search recipe by name

Also added unit tests for a bunch of other reusable react components. Refactored Create cookbook to use React server actions for integration to create cookbook backend.

![image](https://github.com/aiverify-foundation/moonshot-ui/assets/130328114/b20f8b80-50a6-41b8-8754-9e7690e50706)


## Motivation and Context

Planned feature

## Type of Change


- feat: A new feature 










## How to Test

Add to new cookbook
1.  Run moonshot with web UI
2.  Go to Benchmarking -> View Recipes
3.  Search recipes using the new search text field
4.  Select recipes by checking recipe checkboxes
5. Go into 'Add to new Cookbook' flow and create a new cookbook
6. Go to Cookbooks list screens and verify Cookbook with the correct recipes were  added
7. Verify in `Benchmarking -> View Cookbooks`
8. Verify in Run benchmark flow (/benchmarking/session/new)

Add to existing cookbook
1.  Run moonshot with web UI
2.  Go to Benchmarking -> View Recipes
3.  Search recipes using the new search text field
4.  Select recipes by checking recipe checkboxes
5. Go into 'Add to Existing Cookbook' flow and add recipes to existing cookbook
6. Go to Cookbooks list screens and verify Cookbook has been updated with the added recipes
7. Verify in `Benchmarking -> View Cookbooks`
8. Verify in Run benchmark flow (/benchmarking/session/new)

Run regression on Create Cookbook Flow
1.  Run moonshot with web UI
2.  Go to `Create Cookbook` from homepage
3.  Verify that the flow works and new cookbook is reflected on other Cookbook list screens

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [ x]  I have tested the changes locally and verified that they work as expected.
- [ x]  I have added appropriate unit tests or functional tests for the changes made.
- [x ]  I have followed the project's coding conventions and style guidelines.
- [x ]  I have rebased my branch onto the latest commit of the main branch.
- [x ]  I have added any necessary dependencies or packages to the project's build configuration.
- [x ]  I have performed a self-review of my own code.
- [x ]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
